src/python/pants/backend/go/tailor_test.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import textwrap
+
+import pytest
+
+from pants.backend.go import module, target_type_rules
+from pants.backend.go.module import ResolvedGoModule, ResolveGoModuleRequest
+from pants.backend.go.tailor import (
+    PutativeGoExternalModuleTargetsRequest,
+    PutativeGoModuleTargetsRequest,
+    PutativeGoPackageTargetsRequest,
+)
+from pants.backend.go.tailor import rules as go_tailor_rules
+from pants.backend.go.target_types import GoExternalModule, GoModule, GoPackage
+from pants.core.goals.tailor import (
+    AllOwnedSources,
+    PutativeTarget,
+    PutativeTargets,
+    PutativeTargetsSearchPaths,
+)
+from pants.core.goals.tailor import rules as core_tailor_rules
+from pants.core.util_rules import external_tool, source_files
+from pants.engine.addresses import Addresses
+from pants.engine.rules import QueryRule
+from pants.engine.target import DependenciesRequest, UnexpandedTargets
+from pants.testutil.rule_runner import RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    rule_runner = RuleRunner(
+        rules=[
+            *core_tailor_rules(),
+            *go_tailor_rules(),
+            *external_tool.rules(),
+            *source_files.rules(),
+            *module.rules(),
+            *target_type_rules.rules(),
+            QueryRule(PutativeTargets, [PutativeGoPackageTargetsRequest, AllOwnedSources]),
+            QueryRule(PutativeTargets, [PutativeGoModuleTargetsRequest, AllOwnedSources]),
+            QueryRule(PutativeTargets, [PutativeGoExternalModuleTargetsRequest, AllOwnedSources]),
```

Comment:
Only these three queries look used: the other ones do not have assocated `rule_runner.request(..)` calls, and thus don't need to be queries.

Reply:
Done.